2 Business layer/CandidateEvaluator.Core/Dispatchers/CommandDispatcher.cs

```diff
@@ -0,0 +1,24 @@
+﻿using CandidateEvaluator.Contract.Commands;
+using CandidateEvaluator.Contract.Dispatchers;
+using CandidateEvaluator.Contract.Handlers;
+using System;
+using System.Threading.Tasks;
+
+namespace CandidateEvaluator.Core.Dispatchers
+{
+    public class CommandDispatcher : ICommandDispatcher
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public CommandDispatcher(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
```

Comment:
This is an antipattern called service locator. Is there a way to get it rid off?

Reply:
I was aware of that, but dispatcher is really small. Another way could be to inject a list of all command handlers and then choose one, but we generic interface so default DI container doesn't allow for that. I'll research if other DI containers have that feature.

Reply:
Factory?

Reply:
If you want to inject a list of handlers this is possible whitout using different IoC

Reply:
Default Asp.Net Core IoC is able to inject a list of generics?

Reply:
[Mentors](https://github.com/devmentors/DNC-DShop.Common/blob/master/src/DShop.Common/Dispatchers/QueryDispatcher.cs) handle that problem in the same way.